bomorinid/wavepool-senior
# Set cover story for front page

## Business requirements
* CMS user can set only 1 cover story at a time
* Site visitor sees the cover story in the cover story section of the front page
* Site visitor sees the three most recent stories excluding the cover story in the top news of the front page, ordered by most recent
* Site visitor all remaining stories in the archive section of the front page, ordered by most recent

## Technical context
* Adds a custom clean to the newspost form to unset previous cover story and set new cover story
* Adds logic to the front page view that selects the correct story as cover story and organizes the remaining stories by top section or archive section

## Manual testing
* In the CMS, set a news post as the cover story on the news post change form
* Open the change form a different news post and set it as cover story, and verify that the previous one is no longer set as the cover story
* On the site front page and verify that they current cover story is displayed as the cover story
* On the site front page, verify the top stories is the next 3 most recent stories and the archive is all remaining stories
* Run the CmsPage.test_only_one_cover_story test and it passes
